src/wp-includes/html-api/class-wp-html-open-elements.php

```diff
@@ -237,23 +301,34 @@ public function has_element_in_scope( string $tag_name ): bool {
 		return $this->has_element_in_specific_scope(
 			$tag_name,
 			array(
-
-				/*
-				 * Because it's not currently possible to encounter
-				 * one of the termination elements, they don't need
-				 * to be listed here. If they were, they would be
-				 * unreachable and only waste CPU cycles while
-				 * scanning through HTML.
-				 */
+				'APPLET',
+				'CAPTION',
+				'HTML',
+				'TABLE',
+				'TD',
+				'TH',
+				'MARQUEE',
+				'OBJECT',
+				'TEMPLATE',
+				// @todo: Support SVG and MathML nodes when support for foreign content is added.
 			)
 		);
 	}
 
 	/**
 	 * Returns whether a particular element is in list item scope.
 	 *
+	 * > The stack of open elements is said to have a particular element
+	 * > in list item scope when it has that element in the specific scope
+	 * > consisting of the following element types:
+	 * >
+	 * >   - All the element types listed above for the has an element in scope algorithm.
+	 * >   - ol in the HTML namespace
+	 * >   - ul in the HTML namespace
+	 *
 	 * @since 6.4.0
 	 * @since 6.5.0 Implemented: no longer throws on every invocation.
+	 * @since 6.7.0 Full implementation.
```

Comment:
Foreign content is not covered yet.
```suggestion
	 * @since 6.7.0 Implementation covering all HTML elements.
```

Reply:
good thought: clarified in 0c9c40f26c8c8749cadf1563cb51357d12852898